From 5742b3288b2494d03149f439182a189ba9330a47 Mon Sep 17 00:00:00 2001
From: Pierre Rossi <pierre.rossi@digia.com>
Date: Mon, 3 Dec 2012 15:16:56 +0100
Subject: [PATCH] [Qt] Use a separate QWEBKITWIDGETS_EXPORT macro for WebKitWidgets

https://bugs.webkit.org/show_bug.cgi?id=103844

Patch by Pierre Rossi <pierre.rossi@digia.com> on 2012-12-03
Reviewed by Simon Hausmann.

Let's follow the common practice and have one export macro per library.
We can't rely on BUILDING_WEBKIT anymore in that case, so we use the
qmake defines instead.

* Api/qwebkitglobal.h: Updated the import/export logic.
* WidgetApi/qgraphicswebview.h: s/QWEBKIT/QWEBKITWIDGETS/
* WidgetApi/qwebframe.h: Ditto.
* WidgetApi/qwebinspector.h: Ditto.
* WidgetApi/qwebpage.h: Ditto.
* WidgetApi/qwebview.h: Ditto.
* WidgetSupport/InitWebKitQt.cpp:
(WebKit::initializeWebKitWidgets): use QWEBKITWIDGETS_EXPORT.
* WidgetSupport/InitWebKitQt.h:
(WebKit): Ditto.

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@136394 268f45cc-cd09-0410-ab3c-d52691b4dbfc
Change-Id: I0030f2f17cddd66aedf24f29323579174e838303
Reviewed-by: Pierre Rossi <pierre.rossi@gmail.com>
---
 Source/WebKit/qt/Api/qwebkitglobal.h            |    8 +++++++-
 Source/WebKit/qt/ChangeLog                      |   22 ++++++++++++++++++++++
 Source/WebKit/qt/WidgetApi/qgraphicswebview.h   |    2 +-
 Source/WebKit/qt/WidgetApi/qwebframe.h          |    4 ++--
 Source/WebKit/qt/WidgetApi/qwebinspector.h      |    2 +-
 Source/WebKit/qt/WidgetApi/qwebpage.h           |    4 ++--
 Source/WebKit/qt/WidgetApi/qwebview.h           |    2 +-
 Source/WebKit/qt/WidgetSupport/InitWebKitQt.cpp |    2 +-
 Source/WebKit/qt/WidgetSupport/InitWebKitQt.h   |    4 +++-
 9 files changed, 40 insertions(+), 10 deletions(-)

diff --git a/Source/WebKit/qt/Api/qwebkitglobal.h b/Source/WebKit/qt/Api/qwebkitglobal.h
index 1d61182..5b866be 100644
--- a/Source/WebKit/qt/Api/qwebkitglobal.h
+++ b/Source/WebKit/qt/Api/qwebkitglobal.h
@@ -27,13 +27,19 @@
 #define QTWEBKIT_VERSION_CHECK(major, minor, patch) ((major<<16)|(minor<<8)|(patch))
 
 #ifndef QT_STATIC
-#  if defined(BUILDING_WEBKIT)
+#  if defined(QT_BUILD_WEBKIT_LIB)
 #      define QWEBKIT_EXPORT Q_DECL_EXPORT
 #  else
 #      define QWEBKIT_EXPORT Q_DECL_IMPORT
 #  endif
+#  if defined(QT_BUILD_WEBKITWIDGETS_LIB)
+#      define QWEBKITWIDGETS_EXPORT Q_DECL_EXPORT
+#  else
+#      define QWEBKITWIDGETS_EXPORT Q_DECL_IMPORT
+#  endif
 #else
 #  define QWEBKIT_EXPORT
+#  define QWEBKIT_EXPORT
 #endif
 
 QWEBKIT_EXPORT QString qWebKitVersion();
diff --git a/Source/WebKit/qt/WidgetApi/qgraphicswebview.h b/Source/WebKit/qt/WidgetApi/qgraphicswebview.h
index 5c5b666..c9e61f7 100644
--- a/Source/WebKit/qt/WidgetApi/qgraphicswebview.h
+++ b/Source/WebKit/qt/WidgetApi/qgraphicswebview.h
@@ -37,7 +37,7 @@ class QWebSettings;
 
 class QGraphicsWebViewPrivate;
 
-class QWEBKIT_EXPORT QGraphicsWebView : public QGraphicsWidget {
+class QWEBKITWIDGETS_EXPORT QGraphicsWebView : public QGraphicsWidget {
     Q_OBJECT
 
     Q_PROPERTY(QString title READ title NOTIFY titleChanged)
diff --git a/Source/WebKit/qt/WidgetApi/qwebframe.h b/Source/WebKit/qt/WidgetApi/qwebframe.h
index f3676fe..481d019 100644
--- a/Source/WebKit/qt/WidgetApi/qwebframe.h
+++ b/Source/WebKit/qt/WidgetApi/qwebframe.h
@@ -63,7 +63,7 @@ class QWebFrameData;
 class QWebHitTestResultPrivate;
 class QWebFrame;
 
-class QWEBKIT_EXPORT QWebHitTestResult {
+class QWEBKITWIDGETS_EXPORT QWebHitTestResult {
 public:
     QWebHitTestResult();
     QWebHitTestResult(const QWebHitTestResult &other);
@@ -104,7 +104,7 @@ private:
     friend class QWebPage;
 };
 
-class QWEBKIT_EXPORT QWebFrame : public QObject {
+class QWEBKITWIDGETS_EXPORT QWebFrame : public QObject {
     Q_OBJECT
     Q_PROPERTY(qreal textSizeMultiplier READ textSizeMultiplier WRITE setTextSizeMultiplier DESIGNABLE false)
     Q_PROPERTY(qreal zoomFactor READ zoomFactor WRITE setZoomFactor)
diff --git a/Source/WebKit/qt/WidgetApi/qwebinspector.h b/Source/WebKit/qt/WidgetApi/qwebinspector.h
index 3733129..c333fa2 100644
--- a/Source/WebKit/qt/WidgetApi/qwebinspector.h
+++ b/Source/WebKit/qt/WidgetApi/qwebinspector.h
@@ -27,7 +27,7 @@
 
 class QWebInspectorPrivate;
 
-class QWEBKIT_EXPORT QWebInspector : public QWidget {
+class QWEBKITWIDGETS_EXPORT QWebInspector : public QWidget {
     Q_OBJECT
 public:
     QWebInspector(QWidget* parent = 0);
diff --git a/Source/WebKit/qt/WidgetApi/qwebpage.h b/Source/WebKit/qt/WidgetApi/qwebpage.h
index 5624b49..6aa303b 100644
--- a/Source/WebKit/qt/WidgetApi/qwebpage.h
+++ b/Source/WebKit/qt/WidgetApi/qwebpage.h
@@ -65,7 +65,7 @@ namespace WebCore {
     struct FrameLoadRequest;
 }
 
-class QWEBKIT_EXPORT QWebPage : public QObject {
+class QWEBKITWIDGETS_EXPORT QWebPage : public QObject {
     Q_OBJECT
 
     Q_PROPERTY(bool modified READ isModified)
@@ -213,7 +213,7 @@ public:
         Geolocation
     };
 
-    class QWEBKIT_EXPORT ViewportAttributes {
+    class QWEBKITWIDGETS_EXPORT ViewportAttributes {
     public:
         ViewportAttributes();
         ViewportAttributes(const QWebPage::ViewportAttributes& other);
diff --git a/Source/WebKit/qt/WidgetApi/qwebview.h b/Source/WebKit/qt/WidgetApi/qwebview.h
index 701fb36..43439df 100644
--- a/Source/WebKit/qt/WidgetApi/qwebview.h
+++ b/Source/WebKit/qt/WidgetApi/qwebview.h
@@ -38,7 +38,7 @@ class QWebPage;
 class QWebViewPrivate;
 class QWebNetworkRequest;
 
-class QWEBKIT_EXPORT QWebView : public QWidget {
+class QWEBKITWIDGETS_EXPORT QWebView : public QWidget {
     Q_OBJECT
     Q_PROPERTY(QString title READ title)
     Q_PROPERTY(QUrl url READ url WRITE setUrl)
diff --git a/Source/WebKit/qt/WidgetSupport/InitWebKitQt.cpp b/Source/WebKit/qt/WidgetSupport/InitWebKitQt.cpp
index 7188603..8d26375 100644
--- a/Source/WebKit/qt/WidgetSupport/InitWebKitQt.cpp
+++ b/Source/WebKit/qt/WidgetSupport/InitWebKitQt.cpp
@@ -30,7 +30,7 @@
 
 namespace WebKit {
 
-Q_DECL_EXPORT void initializeWebKitWidgets()
+QWEBKITWIDGETS_EXPORT void initializeWebKitWidgets()
 {
     static bool initialized = false;
     if (initialized)
diff --git a/Source/WebKit/qt/WidgetSupport/InitWebKitQt.h b/Source/WebKit/qt/WidgetSupport/InitWebKitQt.h
index fbb2d18..772c960 100644
--- a/Source/WebKit/qt/WidgetSupport/InitWebKitQt.h
+++ b/Source/WebKit/qt/WidgetSupport/InitWebKitQt.h
@@ -21,9 +21,11 @@
 #ifndef InitWebKitQt_h
 #define InitWebKitQt_h
 
+#include <qwebkitglobal.h>
+
 namespace WebKit {
 
-Q_DECL_EXPORT void initializeWebKitWidgets();
+QWEBKITWIDGETS_EXPORT void initializeWebKitWidgets();
 
 }
 #endif
-- 
1.7.1

